ivy/functional/frontends/torch/spectra_ops.py

```diff
@@ -0,0 +1,440 @@
+    return ivy.lcm(input, other, out=out)
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
+def einsum(equation, *operands):
+    if len(operands) == 1 and isinstance(operands[0], (list, tuple)):
+        operands = operands[0]
+    return ivy.einsum(equation, *operands)
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
+def cross(input, other, dim=None, *, out=None):
+    if dim is None:
+        dim = -1
+    input, other = promote_types_of_torch_inputs(input, other)
+    return ivy.cross(input, other, axisa=-1, axisb=-1, axisc=-1, axis=dim, out=out)
+
+
+@to_ivy_arrays_and_back
+def gcd(input, other, *, out=None):
+    return ivy.gcd(input, other, out=out)
+
+
+@with_unsupported_dtypes({"2.0.1 and below": ("float16", "bfloat16")}, "torch")
+@to_ivy_arrays_and_back
+def tensordot(a, b, dims=2, out=None):
+    a, b = promote_types_of_torch_inputs(a, b)
+    return ivy.tensordot(a, b, axes=dims, out=out)
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes(
+    {"2.0.1 and below": ("int8", "float16", "bfloat16", "bool")}, "torch"
+)
+def diff(input, n=1, dim=-1, prepend=None, append=None):
+    return ivy.diff(input, n=n, axis=dim, prepend=prepend, append=append, out=None)
+
+
+@to_ivy_arrays_and_back
+def broadcast_shapes(*shapes):
+    return ivy.broadcast_shapes(*shapes)
+
+
+@to_ivy_arrays_and_back
+def atleast_2d(*tensors):
+    return ivy.atleast_2d(*tensors)
+
+
+@to_ivy_arrays_and_back
+def searchsorted(
+    sorted_sequence,
+    values,
+    /,
+    *,
+    out_int32=False,
+    right=False,
+    side="left",
+    out=None,
+    sorter=None,
+):
+    if right and side == "left":
+        raise ivy.exceptions.IvyError(
+            "side and right can't be set to opposites, got side of left"
+            " while right was True"
+        )
+    if right:
+        side = "right"
+    ret = ivy.searchsorted(sorted_sequence, values, side=side, out=out, sorter=sorter)
+    if out_int32:
+        ret = ivy.astype(ret, "int32")
+    return ret
+
+
+@to_ivy_arrays_and_back
+def atleast_3d(*tensors):
+    return ivy.atleast_3d(*tensors)
+
+
+@to_ivy_arrays_and_back
+def diag(input, diagonal=0, *, out=None):
+    return ivy.diag(input, k=diagonal)
+
+
+@to_ivy_arrays_and_back
+def clone(input):
+    return ivy.copy_array(input)
+
+
+@to_ivy_arrays_and_back
+def cov(input, /, *, correction=1, fweights=None, aweights=None):
+    return ivy.cov(input, ddof=correction, fweights=fweights, aweights=aweights)
+
+
+@with_supported_dtypes(
+    {"2.0.1 and below": ("complex64", "complex128")},
+    "torch",
+)
+@to_ivy_arrays_and_back
+def view_as_real(input):
+    if not ivy.is_complex_dtype(input):
+        raise ivy.exceptions.IvyError(
+            "view_as_real is only supported for complex tensors"
+        )
+    re_part = ivy.real(input)
+    im_part = ivy.imag(input)
+    return ivy.stack((re_part, im_part), axis=-1)
+
+
+@to_ivy_arrays_and_back
```

Comment:
This should be the only function you should implement, can you please un-commit all the other functions, they are all present in `miscellaneous_ops.py`thanks!

Reply:
Okay, I will make the changes as directed.
Thank you very much for the update.